dts/bindings/led_strip/ti,tlc59731.yaml

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2024 Javad Rahimipetroudi <javad.rahimipetroudi@mind.be>
+# SPDX-License-Identifier: Apache-2.0
+
+compatible: "ti,tlc59731"
+description: TLC59731 RGB LED Controller
+
+properties:
```

Comment:
You should add a `chain-length` property, and some support in the driver as well.

Reply:
It has data in and data out though, which is the same as e.g. ws2812, so why would it not support a chain length? If a board had 3 of these in a chain, how does someone set them?

Reply:
> It has data in and data out though, which is the same as e.g. ws2812, so why would it not support a chain length? If a board had 3 of these in a chain, how does someone set them?

Based on EasySet protocol, when we are going to update an specific LED, we have to update the whole chain. So the user knows the whole chain length. For the driver, it should only read the data in a chain and then send it over the channel. 
For example, consider we have a chain of LEDs with this type of driver. (from datasheet)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/5605503/efe7c418-6a8c-4947-a2a6-493d5b9e06e6)
If you want to set or update the values of the LEDs, you need to define an array with a length equal to the number of LEDs and fill it with the desired values.
Then, the chain is updated based on the following procedure 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/5605503/4af8a736-4512-4305-ba95-5e9b92783212)
or this:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/5605503/35015029-cef3-49d7-97b0-4d654a4ee59f)

Exactly in which section, the `chain-length`  could be used?
I have also reviewed the `ws2812` gpio based. the `chain-length` is not used in it. But used in its `SPI` implementation to define the SPI buffer as follows:
```c
#define WS2812_SPI_NUM_PIXELS(idx) \
        (DT_INST_PROP(idx, chain_length))
#define WS2812_SPI_BUFSZ(idx) \
        (WS2812_NUM_COLORS(idx) * 8 * WS2812_SPI_NUM_PIXELS(idx))
```
So, I think it is not useful to add it. At least for gpio based driver.

Reply:
This is exactly what chain length is used for, it specifies how many grouped LEDs on the bus there are, taking this picture https://www.witop-tech.com/wp-content/uploads/2017/07/WS2818-point-led-strip.jpg the chain length is 9

Reply:
> Exactly in which section, the `chain-length` could be used? I have also reviewed the `ws2812` gpio based. the `chain-length` is not used in it. But used in its `SPI` implementation to define the SPI buffer as follows:

It is used by the user application to allocate the buffer to pass to the `update_rgb` and `update_channels` functions. If you don't describe in DT how many LEDs are available in the strip, how the user and driver will guess ?

Reply:
As I searched for this variable, it is in SPI and I2S implementation of WS2812 driver to specify the driver's buffer length:
```bash
zephyr/drivers/led_strip$ grep -d recurse "chain_length"

led_strip/ws2812_i2s.c:#define WS2812_I2S_NUM_PIXELS(idx) (DT_INST_PROP(idx, chain_length))
led_strip/ws2812_spi.c:	(DT_INST_PROP(idx, chain_length))

````
```c
#define WS2812_I2S_BUFSIZE(idx)                                                                    \
        (((WS2812_NUM_COLORS(idx) * WS2812_I2S_NUM_PIXELS(idx)) +                                  \
          WS2812_I2S_PRE_DELAY_WORDS + WS2812_RESET_DELAY_WORDS(idx)) * 4)
```
 In the `gpio` implementation of TLC59731 driver (also `ws2812_gpio` driver ),  the driver doesn't use any internal buffer. It simply gets the data from the user buffer and then passes it to the device as a bit stream. 
However, I will put it in the DT to help user to specify their user-space buffer length.